app/models/compound_aliquot.rb

```diff
@@ -54,13 +54,34 @@ def source_aliquots_have_same_project
     errors.add(:base, "#{MULTIPLE_PROJECTS_ERROR_MSG}: #{component_samples.map(&:name)}")
   end
 
+  def find_or_create_compound_sample(source_aliquots)
+    #  Check if a compound sample already exists with the source_aliquot samples
+    existing_compound_sample = find_compound_sample_for_component_samples(source_aliquots)
+
+    @compound_sample = existing_compound_sample || create_compound_sample
+  end
+
+  # Due to previous implementation, there may be multiple compound samples with the provided component samples.
+  # NPG have confirmed we do not need to fix the data where there are multiple compound samples with
+  # the same component samples
+  def find_compound_sample_for_component_samples(source_aliquots)
+    # Get all the component samples
+    component_samples = source_aliquots.map(&:sample)
```

Comment:
Think you don't need to pass `source_aliquots` into this method, and can just use `component_samples` straight away, as you set it in the `initialize` method?

Reply:
Yes of course. Thanks. Wow, my rails knowledge has really dropped!